infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Workflow.java

```diff
@@ -0,0 +1,28 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import java.util.List;
+
+/**
+ *
+ * A workflow is an abstract container that specifies a series of ordered steps that achieves a
+ * multi-step operation. For example, adding a new node to the cluster, which requires several
+ * steps to complete.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public interface Workflow {
+
+    /**
+     * Return the name of this workflow
+     * @return workflow's name
+     */
+    String getName();
+
+    /**
+     * Returns the ordered that are associated with
+     * this workflow.
+     * @return List of actions
+     */
+    List<Action> getActions();
```

Comment:
I don't think this answers @medhavidhawan 's question. Is there a need for actions to ever refer to the workflow they are running in or not?

Reply:
Workflows "own" actions that are not shared with other workflows and every workflow can refer to its own actions since they maintain action references via getActions.